executor/load_data.go

```diff
@@ -404,9 +404,43 @@ func (e *LoadDataInfo) getValidData(prevData, curData []byte) ([]byte, []byte) {
 	return nil, curData
 }
 
+func (e *LoadDataInfo) isInQuoter(bs []byte) bool {
+	inQuoter := false
+
+	type termType int
+	const (
+		none termType = iota
+		enclosed
+		escaped
+	)
+
+	cmpTerm := func(bs byte) (typ termType) {
+		if bs == e.FieldsInfo.Enclosed {
+			return enclosed
+		}
+		if bs == e.FieldsInfo.Escaped {
+			return escaped
+		}
+		return none
+	}
+
+	for i := 0; i < len(bs); i++ {
+		switch cmpTerm(bs[i]) {
```

Comment:
Why not switch `bs[i]` directly? Then we needn't `cmpTerm` and `termType`.

Reply:
Users can specify any valid characters as the enclosed character and the escaped character. If use switch bs[i] directly, we must specify a const character after 'case'. however the character should be specified by users.

Reply:
> Why not switch `bs[i]` directly? Then we needn't `cmpTerm` and `termType`.

Users can specify any valid characters as the enclosed character and the escaped character. If use switch bs[i] directly, we must specify a const character after 'case'. however the character should be specified by users.

Reply:
I don't understand, I mean using the following code instead of line410-435.
```
	for i := 0; i < len(bs); i++ {
		switch bs[i] {
		case e.FieldsInfo.Enclosed:
			inQuoter = !inQuoter
		case e.FieldsInfo.Escaped:
			i++
		default:
		}
	}
```

Reply:
> I don't understand, I mean using the following code instead of line410-435.
> 
> ```
> 	for i := 0; i < len(bs); i++ {
> 		switch bs[i] {
> 		case e.FieldsInfo.Enclosed:
> 			inQuoter = !inQuoter
> 		case e.FieldsInfo.Escaped:
> 			i++
> 		default:
> 		}
> 	}
> ```

Ok, It seems batter. I have modified the code.